lib/internal/http2/compat.js

```diff
@@ -705,12 +705,15 @@ class Http2ServerResponse extends Stream {
   }
 }
 
-function onServerStream(ServerRequest, ServerResponse,
+function onServerStream(ServerRequest, ServerResponse, kStreamEventsComposite,
                         stream, headers, flags, rawHeaders) {
   const server = this;
   const request = new ServerRequest(stream, headers, undefined, rawHeaders);
   const response = new ServerResponse(stream);
 
+  stream[kStreamEventsComposite].push(request);
+  stream[kStreamEventsComposite].push(response);
```

Comment:
Instead of adding things to an array, I would add an event listener to `'timeout'` on the stream.

Reply:
@mcollina as I described in the 2nd comment [here](https://github.com/nodejs/node/pull/20918#issuecomment-392104693), if we add a `timeout` listener to an `EventEmitter`, the checks with `if (...emit(...))` will not work, because they all are always true then.